tokio/src/runtime/context.rs

```diff
@@ -101,6 +104,18 @@ cfg_rt! {
 
     pub(crate) struct DisallowBlockInPlaceGuard(bool);
 
+    pub(crate) fn set_current_task_id(id: Option<Id>) {
+        let _ = CONTEXT.try_with(|ctx| ctx.current_task_id.replace(id));
+    }
+
+    #[track_caller]
+    pub(crate) fn current_task_id() -> Id {
+        match CONTEXT.try_with(|ctx| ctx.current_task_id.get()) {
+            Ok(Some(id)) => id,
+            _ => panic!("can't get a task id when not inside a task"),
```

Comment:
will this panic ever be encountered by a user, or does it indicate an internal bug? if it's going to be encountered if user code tries to get the current task ID outside of a task, it should probably explain a bit more about what happened. on the other hand, if it indicates a bug in Tokio, we should probably use `unreachable!` and state that this is a Tokio bug that should be reported.

Reply:
Yeah, you will hit this if you call `task::id()` from outside of a task.